lib/Graph.pm

```diff
@@ -1825,6 +1825,28 @@ sub directed_copy {
 
 *directed_copy_graph = \&directed_copy;
 
+sub is_bipartite {
+    my $g = shift;
+
+    my $is_bipartite = 1;
```

Comment:
Does this not have the effect that an empty and/or unconnected and/or single-vertex graph is considered bipartite? That's probably worth testing for, and returning `undef`, and documenting.

Reply:
According to [Wikipedia](https://en.wikipedia.org/w/index.php?title=Bipartite_graph&oldid=1225953055),

> In the mathematical field of graph theory, a bipartite graph (or bigraph) is a graph whose vertices can be divided into two disjoint and independent sets U and V, that is, every edge connects a vertex in U to one in V. Vertex sets U and V are usually called the parts of the graph. Equivalently, a bipartite graph is a graph that does not contain any odd-length cycles.

I prefer the equivalent definition based on the absence of odd-length cycles. Based on it, graphs without vertices or edges are considered bipartite. Similarly, a graph of two or more connected component is bipartite iff at least one of its components is bipartite. I believe the current approach handles these cases.

Let me know if you want me to find a reference better than Wikipedia.

But your comment got me thinking about other types of graphs, namely directed ones. I can think of a case where the current approach would emit a false negative. Therefore I better limit the subroutine to undirected graphs for now.

Reply:
Fair enough. I have added that to the doc for you.